IdentityCore/src/cache/token/MSIDCredentialCacheItem.m

```diff
@@ -302,4 +302,17 @@ - (BOOL)isTombstone
     return [self.secret isEqualToString:@"<tombstone>"];
 }
 
+- (BOOL)storeAppKey:(NSString *)cacheKeyString
+{
+    if ([NSString msidIsStringNilOrBlank:cacheKeyString]) return NO;
+    
+    NSArray *components = [cacheKeyString componentsSeparatedByString:@"|"];
```

Comment:
Since the logic of appending app key lives on keychain token cache layer, I'd prefer the logic of reading app key also be there and MSIDCredentialCacheItem can continue being a more dummy object then.

Reply:
agree, moved to keychain token cache :)